app/controllers/users_controllers.rb

```diff
@@ -0,0 +1,35 @@
+class UsersControllers < ApplicationController
+  before_action :require_admin, only: [:index, :show, :new, :create, :destroy]
+  before_action :require_user, only: [:edit, :update]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    authorize @user
+    if @user.update_attributes(secure_params)
+      redirect_to users_path, :notice => "User updated."
+    else
+      redirect_to users_path, :alert => "Unable to update user."
+    end
+  end
+
+  def destroy
+    user = User.find(params[:id])
+    authorize user
+    user.destroy
+    redirect_to users_path, :notice => "User deleted."
```

Comment:
The same: Logic blocks, move find user into method, don't use rocket style in hash. Use {obj: 1}